lib/ansible/modules/cloud/amazon/ec2_vpc_vpn.py

```diff
@@ -0,0 +1,499 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: ec2_vpc_vpn
+short_description: create, modify, and delete VPN connections
+description:
+  - This module creates, modifies, and deletes VPN connections. Idempotence is achieved by using the filters
+    options or specifying the VPN connection identifier.
+version_added: "2.4"
+author: "Sloane Hertel (@s-hertel)"
+options:
+  state:
+    description:
+      - the desired state of the vpn connection.
+    choices: ['present', 'absent']
+    default: present
+    required: no
+  customer_gateway_id:
+    description:
+      - The ID of the customer gateway.
+  connection_type:
+    description:
+      - The type of VPN connection
+    choices: ['ipsec.1']
+    default: ipsec.1
+  vpn_gateway_id:
+    description:
+      - The ID of the virtual private gateway.
+  vpn_connection_id:
+    description:
+      - The ID of the vpn connection. Required to modify or delete a connection if the filters option does not provide a unique match.
+  tags:
+    description:
+      - Tags to attach to the vpn connection
+  purge_tags:
+    description:
+      - Whether or not to delete vpn connections tags that are associated with the connection but not specified in the task
+    type: bool
+    default: false
+  static_only:
+    description:
+      - Indicates whether the VPN connection uses static routes only. Static routes must be used for devices that don't support BGP.
+    default: False
+    required: no
+  filters:
+    description:
```

Comment:
Yes, that's right. I am going to document this a little better with `suboptions:` and I'm still debating the filter names because `route.destination-cidr-block` doesn't feel very user friendly. I'll probably rename some things. Also need to verify if these are the correct types for each option and add a few examples.